### ISSUE TITLE
Fine-tuning and linear evaluation

### ISSUE THREAD

**Issue by besaman** [IcarusWizard/MAE#11]
Thank you so much for this simple yet effective code!

Excuse me because I'm still new to this. My questions are about [train_classifier.py](https://github.com/IcarusWizard/MAE/blob/main/train_classifier.py) code. 
1- Does it do fine-tuning or linear evaluation? 
2- Assuming it does one of them, how to toggle to do the other? .. what should I change?
3- does it use the encoder or the decoder to do this?

Excuse my basic questions, but you answer will be really appreciated.

**Comment by IcarusWizard** [comment]
Hi @besaman,

1. The code is doing fine-tuning.
2. I think the easiest way to switch to linear evaluation is to add before [line 47](https://github.com/IcarusWizard/MAE/blob/main/train_classifier.py#L47) `model.encoder.requires_grad_(False)`, which will turn off the gradient for the encoder.
3. Only the encoder is used when training the classifier. 

Hope these help.

**Comment by besaman** [comment]
> Hope these help.

Very much .. thank you!

Last question: you mentioned that you are using vit-tiny instead of vit-base .. how to change it to vit-base?

**Comment by IcarusWizard** [comment]
Just change `emb_dim=768` and `encoder_head=12`.

**Comment by besaman** [comment]
Thank you so much!
